File: ui/src/classes/Building.ts
import Address from "@/classes/Address";

export default class Building{
    public id: string;
    public name: string;
    public address: Address;

    constructor(id: string, name: string, address: Address){
        this.id = id;
        this.name = name;
        this.address = address;
    }
}

File: api-gateway/Models/ServiceModels/Location/Building.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace api_gateway.Models.ServiceModels.Location
{
    public class Building
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public Address Address { get; set; }
    }
}
